src/server/transaction.h

```diff
@@ -213,7 +213,8 @@ class Transaction {
   // or b) tp is reached. If tp is time_point::max() then waits indefinitely.
   // Expects that the transaction had been scheduled before, and uses Execute(.., true) to register.
   // Returns false if timeout occurred, true if was notified by one of the keys.
-  facade::OpStatus WaitOnWatch(const time_point& tp, WaitKeysProvider cb, KeyReadyChecker krc);
+  facade::OpStatus WaitOnWatch(const time_point& tp, WaitKeysProvider cb, KeyReadyChecker krc,
+                               ConnectionContext* cntx);
```

Comment:
In the future I'd suggest not passing ConnectionContext* but instead a FlagPointers { bool* paused, bool* blocked }, so we don't have this general type here in transaction code. Just a nit here, I can update it later